lib/fs/basicfs.go

```diff
@@ -71,11 +71,32 @@ func (f *BasicFilesystem) DirNames(name string) ([]string, error) {
 }
 
 func (f *BasicFilesystem) Open(name string) (File, error) {
-	return os.Open(name)
+	fd, err := os.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	return fsFile{fd}, err
 }
 
 func (f *BasicFilesystem) Create(name string) (File, error) {
-	return os.Create(name)
+	fd, err := os.Create(name)
+	if err != nil {
+		return nil, err
+	}
+	return fsFile{fd}, err
+}
+
+// fsFile implements the fs.File interface on top of an os.File
+type fsFile struct {
+	*os.File
+}
+
+func (f fsFile) Stat() (FileInfo, error) {
+	info, err := f.File.Stat()
+	if err != nil {
```

Comment:
I guess you could pass a handle to the underlying fs, and it could do f.fs.Stat()